nav2_util/include/nav2_util/validate_messages.hpp

```diff
@@ -0,0 +1,106 @@
+//  Check recieved message is safe or not for the nav2-system
+//  For each msg-type known in nav2, we could check it as following:
+//  if(!validateMsg()) RCLCPP_ERROR(,"malformed msg. Rejecting.")
+//
+//  Workflow of validateMsg():
+//     if here's a sub-msg-type in the recieved msg,
+//        the content of sub-msg would be checked as sub-msg-type
+//     then, check the whole recieved msg.
+//
+//  Following conditions are involved in check:
+//     1> Value Check: to avoid damaged value like like `nan`, `INF`
+//     2> Logic Check: to avoid value with bad logic, like the size of `map` should be equal to `height*width`
+//     3> Any other needed condition could be joint here in future
+
+#ifndef validate_message_HPP
+#define validate_message_HPP
+
+
+#include "nav_msgs/msg/occupancy_grid.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+
+
+namespace nav2_util
+{
+
+const int NSEC_PER_SEC = 1e9;  // 1 second = 1e9 nanosecond
+
+bool validateMsg(const builtin_interfaces::msg::Time_<std::allocator<void>> & msg)
+{
+  // check item
+  if (msg.sec < 0) {return false;}                                      // invalid second-stamp
+  if (msg.nanosec < 0 || msg.nanosec >= NSEC_PER_SEC) {return false;}   // invalid nanosec-stamp
+  return true;
+}
+
+bool validateMsg(const std_msgs::msg::Header_<std::allocator<void>> & msg)
+{
+  //  check sub-type
+  if (!validateMsg(msg.stamp)) {return false;}
+  // msg.frame_id :  @todo any check for it ?
```

Comment:
Check if frame id is an empty string

Reply:
I was concerned about the possibility of a default empty string for the frame_id. That's why I didn't include that check to ensure the generality of this function's functionality.

So can we be certain that empty frame_id must be definitely invalid ?

Reply:
That is the default, which is why we need to check that it contains some actual valid information 

Empty frame is definitely invalid for the majority of things that include a header. There are some special cases to that, but not anywhere that we're talking about using this